### PR TITLE
Consider JobType when resolving an institution

### PR DESCRIPTION
**Pull request by Craiting**
Issue: https://universalconnect.atlassian.net/browse/UCP-163

Resolver checks jobtype to see if provider supports it.

### CODE REVIEW

File: apps/server/src/services/institutionResolver.ts [hunk 1]
```diff
@@ -88,18 +91,61 @@ export async function resolveInstitutionProvider(
...
   }
 
   return providers as Provider[]
 }
+
+function providerSupportsJobType(
+  institutionAttributes: InstitutionProvider | undefined,
+  jobType: JobType
+): boolean {
+  if (institutionAttributes == null) {
```
Comment (tphalp) [inline apps/server/src/services/institutionResolver.ts:120]: I feel like a better check here would be: 

`if (!institutionAttributes) {`

Checking for null is different than a variable or parameter that hasn't been passed in, which generally would be `undefined`. 

If you know that the possible values of `institutionAttributes` will sometimes be `null`, then the check you have will work as expected.

**Review by tphalp — APPROVED** [review]
Just one comment, but not critical.

File: apps/server/src/services/institutionResolver.ts (hunk 1 above)
Comment (tphalp) [reply apps/server/src/services/institutionResolver.ts:120]: Actually, because you are doing a loose comparison (`==`), it will probably work as expected.

Comment (wesrisenmay-mx) [reply apps/server/src/services/institutionResolver.ts:120]: Agreed on the !

**Review by wesrisenmay-mx — APPROVED** [review]

**Pull request merged**